hibernate-core/src/main/java/org/hibernate/hql/internal/ast/tree/SelectClause.java

```diff
@@ -193,7 +193,7 @@ public void initializeExplicitSelectClause(FromClause fromClause) throws Semanti
 				FromElement fromElement = (FromElement) iterator.next();
 
 				if ( fromElement.isFetch() ) {
-					FromElement origin = null;
+					FromElement origin;
```

Comment:
To ensure `origin` gets initialized if/when code gets changed, the declaration should be:

{{final FromElement origin;}}

Reply:
Accessing an uninitialized local variable will result in a compile-time error so making this final does not make any difference I think - but I'll change it.